File: Add Code Here/C++/leetcode/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n = nums.size();
        
        unordered_map <int,int> mp;
        vector<int> res;
        
        
        for (int i=0; i<n; i++) {
            if (mp.find(nums[i])!=mp.end()) {
                res = {mp[nums[i]],i};
                return res;
            }
            else {
                mp[target-nums[i]] = i;
            }
        }
        return res;
    }
};

File: Add Code Here/two-sum/twosum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int[]ans=new int[2];
        int val=0;
        for(int i=0; i<nums.length; i++){
           val = target - nums[i];
           if(mp.containsKey(val)==false) mp.put(nums[i],i);
           else{
               ans[0] = mp.get(val);
               ans[1] = i;
               return ans;

           }
        }
        return ans;
    }
}
